src/javascript/app/pages/user/metatrader/metatrader.ui.js

```diff
@@ -584,6 +587,64 @@ const MetaTraderUI = (() => {
         });
     };
 
+    const setDemoTopupStatus = (is_topped_up = false) => {
+        const el_demo_topup_btn  = getElementById('demo_topup_btn');
+        const el_demo_topup_info = el_demo_topup_btn.previousSibling;
+        const el_loading         = el_demo_topup_btn.parentElement.firstChild;
+        const acc_type           = Client.get('mt5_account');
+        const is_demo            = accounts_info[acc_type].is_demo;
+        const topup_info_text    = localize('You can top up your demo account with an additional [_1] if your balance falls below [_2].', [`${MetaTraderConfig.getCurrency(acc_type)} 10,000.00`, `${MetaTraderConfig.getCurrency(acc_type)} 1,000.00`]);
+        const topup_btn_text     = localize('Get [_1]', `${MetaTraderConfig.getCurrency(acc_type)} 10,000.00`);
+
+        el_loading.setVisibility(0);
+        el_demo_topup_info.innerText = topup_info_text;
+        el_demo_topup_btn.firstChild.innerText = topup_btn_text;
+
+        if (is_demo) {
+            const balance     = +accounts_info[acc_type].info.balance;
+            const min_balance = 1000;
+
+            if (balance < min_balance && !is_topped_up) {
+                enableDemoTopup(true);
+            } else {
+                enableDemoTopup(false);
+            }
+        }
+    };
+
+    const enableDemoTopup = (is_enabled) => {
+        const el_demo_topup_btn = getElementById('demo_topup_btn');
+
+        if (is_enabled) {
+            el_demo_topup_btn.addEventListener('click', topup_demo);
+            el_demo_topup_btn.classList.add('button');
+            el_demo_topup_btn.classList.remove('button-disabled');
+            el_demo_topup_btn.previousSibling.setVisibility(0);
+        } else {
+            el_demo_topup_btn.removeEventListener('click', topup_demo);
+            el_demo_topup_btn.classList.add('button-disabled');
+            el_demo_topup_btn.classList.remove('button');
+            el_demo_topup_btn.previousSibling.setVisibility(1);
+        }
+    };
+
+    const setTopupLoading = (is_loading, is_topped_up) => {
+        const el_demo_topup_btn  = getElementById('demo_topup_btn');
+        const el_demo_topup_info = getElementById('demo_topup_info');
+        const el_loading         = getElementById('demo_topup_loading');
+
+        if (is_loading) {
+            el_demo_topup_btn.setVisibility(0);
+            el_demo_topup_info.setVisibility(0);
+            el_loading.setVisibility(1);
+        } else {
+            el_demo_topup_btn.setVisibility(1);
+            el_demo_topup_info.setVisibility(1);
```

Comment:
I meant something more like `el_demo_topup_info.setVisibility(!is_loading);` to reduce repetition, but if you think this is more readable let's keep it.